AndroidHomeApplication/app/src/main/java/com/example/androidhomeapplication/ui/moviesList/MoviesListViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.androidhomeapplication.ui.moviesList
+
+import androidx.lifecycle.*
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.cachedIn
+import com.example.androidhomeapplication.data.constans.Constants
+import com.example.androidhomeapplication.data.models.Movie
+import com.example.androidhomeapplication.data.paging.MoviesPagingSource
+import com.example.androidhomeapplication.data.repository.MoviesRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.debounce
+
+class MoviesListViewModel(
+    private val moviesRepository: MoviesRepository
+) : ViewModel() {
+
+    private var searchPagingSource = MoviesPagingSource(moviesRepository, null)
+
+    val movieItems: Flow<PagingData<Movie>> by lazy {
+        Pager(PagingConfig(pageSize = Constants.DEFAULT_ITEM_PER_PAGE)) {
+            searchPagingSource
+        }.flow.cachedIn(viewModelScope).debounce(500)
```

Comment:
debounce not here
but on search

maybe will be helpful:
https://blog.mindorks.com/instant-search-using-kotlin-flow-operators